pkg/operators/matchers/compile.go

```diff
@@ -5,12 +5,23 @@ import (
 	"fmt"
 	"regexp"
 	"strings"
+	"sync"
+	"sync/atomic"
 
 	"github.com/Knetic/govaluate"
 
 	"github.com/projectdiscovery/nuclei/v3/pkg/operators/common/dsl"
 )
 
+var (
+	regexCache        sync.Map // map[string]*regexp.Regexp
+	dslCache          sync.Map // map[string]*govaluate.EvaluableExpression
```

Comment:
same here as previous. can we share these somehow to not duplicate

Reply:
done